client/tso_dispatcher.go

```diff
@@ -709,62 +712,67 @@ func (c *tsoClient) processRequests(
 
 	requests := tbc.getCollectedRequests()
 	count := int64(len(requests))
+	reqKeyspaceGroupID := c.svcDiscovery.GetKeyspaceGroupID()
 	respKeyspaceGroupID, physical, logical, suffixBits, err := stream.processRequests(
-		c.svcDiscovery.GetClusterID(), c.svcDiscovery.GetKeyspaceID(), c.svcDiscovery.GetKeyspaceGroupID(),
+		c.svcDiscovery.GetClusterID(), c.svcDiscovery.GetKeyspaceID(), reqKeyspaceGroupID,
 		dcLocation, requests, tbc.batchStartTime)
 	if err != nil {
 		c.finishRequest(requests, 0, 0, 0, err)
 		return err
 	}
-	// `logical` is the largest ts's logical part here, we need to do the subtracting before we finish each TSO request.
+	curTSOInfo := &tsoInfo{
+		tsoServer:           stream.getServerAddr(),
+		reqKeyspaceGroupID:  reqKeyspaceGroupID,
+		respKeyspaceGroupID: respKeyspaceGroupID,
+		respReceivedAt:      time.Now(),
+		physical:            physical,
+		logical:             logical, // `logical` is the returned largest ts's logical part
```

Comment:
Here is different from the previous implementation. Should we use firstLogical?

Reply:
it's same logic.
after:
firstLogical := tsoutil.AddLogical(logical, -count+1, suffixBits)
largestLogical := tsoutil.AddLogical(firstLogical, count-1, suffixBits)

logical == largestLogical